### PR TITLE
Rewrote the "Templates" guide

### PR DESCRIPTION
**Pull request by webmozart**
| Q | A
| --- | ---
| Fixed tickets | -
| License | MIT

#### What's in this PR?

In this PR, I rewrote the "Templates" section, tried to make it more easily understandable and added missing information, examples and images.

### CODE REVIEW

File: book/templates.rst [hunk 1]
```diff
@@ -1,200 +1,721 @@
...
+| |tag_list|              | autocomplete input for entering and adding  | array of strings                        |
+|                         | tags                                        |                                         |
++-------------------------+---------------------------------------------+-----------------------------------------+
+| |category_list|         | autocomplete input for entering and adding  | array of strings                        |
+|                         | tags                                        |                                         |
++-------------------------+---------------------------------------------+-----------------------------------------+
+| |media_selection|       | widget for selecting media (images,         | array containing another array with     |
+|                         | documents)                                  | urls for every format                   |
++-------------------------+---------------------------------------------+-----------------------------------------+
+| |contact_selection|     | widget for selecting contacts               | ?                                       |
```
Comment (webmozart) [inline book/templates.rst]: @danrot @wachterjohannes What's the value here?

File: book/templates.rst [hunk 2]
```diff
@@ -1,200 +1,721 @@
...
++-------------------------+---------------------------------------------+-----------------------------------------+
+| |snippet|               | widget for selecting snippets               | ?                                       |
```
Comment (webmozart) [inline book/templates.rst]: @danrot @wachterjohannes And here?

File: book/templates.rst (hunk 1 above)
Comment (danrot) [inline book/templates.rst]: An array representation of the contact

File: book/templates.rst (hunk 2 above)
Comment (danrot) [inline book/templates.rst]: An array representation of the snippet

File: book/templates.rst [hunk 3]
```diff
@@ -1,200 +1,721 @@
...
+            <!-- ... -->
+
+            <property name="endDate" type="date">
+                <meta>
+                    <!-- ... -->
+
+                    <info_text lang="en">
+                        If the same as the start date, the event is treated as
+                        one-day event.
+                    </title>
```
Comment (danrot) [inline book/templates.rst]: `</info_text>`

File: book/templates.rst (hunk 1 above)
Comment (webmozart) [inline book/templates.rst]: I guess an array of arrays? (multiple selection possible?)

File: book/templates.rst (hunk 2 above)
Comment (webmozart) [inline book/templates.rst]: same here?

File: book/templates.rst (hunk 1 above)
Comment (danrot) [inline book/templates.rst]: Yeah, you are right 😄

File: book/templates.rst (hunk 2 above)
Comment (danrot) [inline book/templates.rst]: Yes.

File: book/templates.rst [hunk 4]
```diff
@@ -1,200 +1,721 @@
...
-                </meta>
+Creating a Custom Template
+--------------------------
 
-                <tag name="sulu.rlp"/>
-            </property>
+In your projects, you will need several templates for different parts of your
+website. The easiest way is to copy and adjust the default template.
+
+The first thing you need to adjust is the ``<key>``. This is the unique
```
Comment (wachterjohannes) [inline book/templates.rst]: this has to be the same as the filename. it is currently a bug but should be mentioned here.

File: book/templates.rst [hunk 5]
```diff
@@ -1,200 +1,721 @@
...
+.. code-block:: xml
+
+    <!-- app/Resources/templates/pages/event.xml -->
+    <?xml version="1.0" ?>
+    <template xmlns="http://schemas.sulu.io/template/template"
+              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">
+        <!-- ... -->
+
+        <view>ClientWebsiteBundle:templates:event</view>
```
Comment (wachterjohannes) [inline book/templates.rst]: we should also mention that the `.<format>.twig` suffix will be added in the rendering process.

File: book/templates.rst [hunk 6]
```diff
@@ -1,200 +1,721 @@
...
+        </properties>
+    </template>
+
+A property has three essential attributes:
+
+* a ``name`` that is unique within a template
+* a ``type`` that defines what kind of content can be stored
+* a ``title`` that is shown in the administration interface
+
+Here is a table with the content types shipped in Sulu core:
```
Comment (wachterjohannes) [inline book/templates.rst]: i think we should mention here the `sulu:content:types:dump` which lists all available content-types in the current system.

File: book/templates.rst (hunk 4 above)
Comment (webmozart) [inline book/templates.rst]: *Very* good to know ;)

File: book/templates.rst (hunk 6 above)
Comment (webmozart) [inline book/templates.rst]: Cool, thanks :)

File: book/templates.rst (hunk 5 above)
Comment (webmozart) [inline book/templates.rst]: isn't that covered by the link to the Symfony doc below?

File: book/templates.rst (hunk 6 above)
Comment (webmozart) [inline book/templates.rst]: Added after the table.

File: book/templates.rst (hunk 4 above)
Comment (webmozart) [inline book/templates.rst]: Fixed.

File: book/templates.rst (hunk 5 above)
Comment (wachterjohannes) [inline book/templates.rst]: in symfony you would have to write the extension. it sulu specific that this will be appended!

**Comment by webmozart** [comment]
:champagne:

**Pull request merged**